packages/utils-tests-helpers/src/hint-runner.ts

```diff
@@ -199,25 +199,68 @@ const validateResults = (t: ExecutionContext<HintRunnerContext>, sources: Map<st
             return;
         }
 
-        const filteredByPosition = filteredByMessage.filter((report) => {
-            if (report.position && location) {
-                let position: ProblemLocation | undefined;
+        const filteredByDocumentation = filteredByMessage.filter((report) => {
+            /*
+             * If the report from the test doesn't ask for documentation,
+             * we don't need to macth it.
+             */
+            if (!report.documentation) {
+                return true;
+            }
 
-                if ('match' in report.position) {
-                    position = findPosition(sources.get(resource) || '', report.position);
-                } else {
-                    position = report.position;
-                }
+            /*
+             * If the report from the test does ask for documentation
+             * but the result doesn't provide it, then it isn't a match.
+             */
+            if (!documentation) {
+                return false;
+            }
+
+            return report.documentation.some((docReport) => {
+                return documentation.some((docResult) => {
+                    return docReport.link === docResult.link &&
+                        docReport.text === docResult.text;
+                });
+            });
+        });
 
-                return position.column === location.column &&
-                    position.line === location.line &&
-                    (!('range' in report.position) || (
-                        position.endLine === location.endLine &&
-                        position.endColumn === location.endColumn));
+        if (filteredByDocumentation.length === 0) {
+            // TODO: Improve message.
```

Comment:
Should we improve it now or just remove this comment?

Reply:
do you have any suggestion for the message? hehehe
If not I can remove the comment

Reply:
It's a test message so I'm fine with it as it is 😊